src/app/reporting/Engine.cpp

```diff
@@ -138,12 +142,15 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeReportIBs(ReportDataMessage::Bu
                     attributeReportIBs.Rollback(attributeBackup);
                     apReadHandler->SetAttributeEncodeState(AttributeValueEncoder::AttributeEncodeState());
 
-                    // Try to encode our error as a status response.
-                    err = attributeReportIBs.EncodeAttributeStatus(pathForRetrieval, StatusIB(err));
-                    if (err != CHIP_NO_ERROR)
+                    if (err != CHIP_ERROR_NO_MEMORY && err != CHIP_ERROR_BUFFER_TOO_SMALL)
                     {
-                        // OK, just roll back again and give up.
-                        attributeReportIBs.Rollback(attributeBackup);
+                        // Try to encode our error as a status response.
+                        err = attributeReportIBs.EncodeAttributeStatus(pathForRetrieval, StatusIB(err));
```

Comment:
If we fail to encode the path-specific error here because we couldn't fit, will it be attempted in the next chunk attempt?

Reply:
yes, I think that logic has not yet been changed, it would be attempted in next chhunk

Reply:
apReadHandler->SetAttributeEncodeState(AttributeValueEncoder::AttributeEncodeState()); which would encode the last attribute data state.
in next chunk, it would load the state, AttributeValueEncoder::AttributeEncodeState encodeState = apReadHandler->GetAttributeEncodeState();